Web/Edubase.Web.UI/Areas/Establishments/Controllers/AcademyOpeningsController.cs

```diff
@@ -49,24 +61,27 @@ public async Task<ActionResult> ManageAcademyOpenings(int skip = 0, string sortB
 
             var to = from.AddYears(30);
 
-
             var property = typeof(EditAcademyOpeningViewModel).GetProperty(sortBy);
 
+            var isUserSecure16To19 = !string.IsNullOrWhiteSpace(isSecure16To19User)
+                                     && bool.Parse(SecureAcademyUtility.DecryptValue(isSecure16To19User));
+
+            if (!string.IsNullOrWhiteSpace(establishmentTypeId))
+                establishmentTypeId = SecureAcademyUtility.DecryptValue(establishmentTypeId);
+
             var estabTypes = await _lookupService.EstablishmentTypesGetAllAsync();
+            estabTypes = SecureAcademyUtility.FilterEstablishmentsByEstablishmentTypeId
+                (estabTypes, establishmentTypeId, isUserSecure16To19);
 
             var result = await _establishmentReadService.SearchAsync(
                 new EstablishmentSearchPayload
                 {
                     Skip = skip,
                     Take = take,
                     SortBy = eSortBy.NameAlphabeticalAZ,
-                    Filters = new EstablishmentSearchFilters
-                    {
-                        OpenDateMin = from,
-                        OpenDateMax = to,
-                        EstablishmentTypeGroupIds = new[] {(int) eLookupEstablishmentTypeGroup.Academies},
-                        StatusIds = new[] {(int) eLookupEstablishmentStatus.ProposedToOpen}
-                    },
+                    Filters =
+                        SecureAcademyUtility.GetEstablishmentSearchFilters(
+                            new GetEstabSearchFiltersParam(from, to, establishmentTypeId, isUserSecure16To19)),
```

Comment:
The SecureAcademyUtility is doing stuff for non-secure academies too here - rename? or extract the construction of the SearchFilters into a factory?

Reply:
True, I think I would just rename it to AcademyUtility